src/taipy/core/_core.py

```diff
@@ -96,6 +97,9 @@ def __manage_version(self, mode, _version_number, force, clean_entities):
 
             version_setter[mode](current_version_number, force)
 
+            if mode == "production":
+                Config._check_one(_MigrationConfigChecker)
```

Comment:
I believe the implementation is too complex. 
1. First, we don't need a generic framework for running one checker and we don't need to delegate this particular checker to the Config. It is just a special case that does not require a generic API on the config side. 
Why not just run:
```
_MigrationConfigChecker(Config._applied_config, IssueCollector())._check()
if len(collector._errors) != 0:
    raise SystemExit("Configuration errors found. Please check the error log for more information.")
```
I guess that would make the PR on Config useless. Right?

2. Second, I also believe that does not fix the problem. If the migration config is incorrect, the version is already created. I believe the code above should be moved up, right after having retrieved the current_version_number, and before setting the version.

Reply:
1. There are 2 reasons for the implementation on Config. 
First, we also need to log the messages. This can be error if the function is not callable, or `_check_if_entity_property_key_used_is_predefined()` found an error, or, of course, not a production version.
Second, this conveniently collects all Issue into the same IssueCollector on Config singleton, which follow the design of a IssueCollector.

2. I agreed. My bad 🥲

Reply:
1. There is no need to reuse the same collector. The "normal" checker already passed without any error. Otherwise, the checker would have raised an error. So we just need to run the _MigrationConfigChecker with a new collector to check for potential new errors or new infos. 

2. I don't get the point.

Reply:
1. Besides from the collector, we will need a different method to log the message as well.
However, I believe this can be done without changing the Config.check() API.

2. I mean the ConfigMigrationChecker need the version to be created first. Then, that version will not be a corrupted version since the user can update the migration config without causing any conflict on the ConfigComparator.